src/test/resources/http-request-errors-config.xml

```diff
@@ -51,6 +51,18 @@
             <on-error-continue type="http:CONNECTIVITY">
                 <set-payload value="#[error.cause.message ++ ' connectivity']"/>
             </on-error-continue>
+            <on-error-continue type="http:GATEWAY_TIMEOUT">
+                <set-payload value="#[error.errorMessage.attributes.statusCode ++ ' Gateway Timeout']"/>
+            </on-error-continue>
+            <on-error-continue type="http:BAD_GATEWAY">
+                <set-payload value="#[error.errorMessage.attributes.statusCode ++ ' Bad Gateway']"/>
+            </on-error-continue>
+            <on-error-continue type="http:CLIENT_ERROR">
```

Comment:
Perhaps `CLIENT_SIDE` and `SERVER_SIDE` would help avoid the `error` redundancy.

Reply:
+1